src/pymortests/lyapunov.py

```diff
@@ -81,25 +80,14 @@ def relative_residual(A, E, B, X, trans=False):
     return res / rhs
 
 
-def _check_availability(lyap_solver):
-    if (lyap_solver.startswith('slycot')
-            and not os.environ.get('DOCKER_PYMOR', False)
-            and not config.HAVE_SLYCOT):
-        pytest.skip('slycot not available')
-    if (lyap_solver.startswith('pymess')
-            and not os.environ.get('DOCKER_PYMOR', False)
-            and not config.HAVE_PYMESS):
-        pytest.skip('pymess not available')
-
-
 @pytest.mark.parametrize('m', m_list)
 @pytest.mark.parametrize('with_E', [False, True])
 @pytest.mark.parametrize('trans', [False, True])
 @pytest.mark.parametrize('n,lyap_solver', chain(product(n_list_small, lyap_dense_solver_list),
                                                 product(n_list_big, lyap_lrcf_solver_list)))
+@skip_if_missing('slycot')
+@skip_if_missing('pymess')
 def test_lrcf(n, m, with_E, trans, lyap_solver):
-    _check_availability(lyap_solver)
```

Comment:
@pmli, is this really what we want? If I understand this correctly, with these changes the test is never executed if either slycot or pymess is missing. Before, only the tests for the respective solver are skipped.

Reply:
Ok, I understood it the same way. Also, I think there are some tests below that do not depend on either `slycot` or `pymess`.
I'm not sure what's the best way to go about it. Having both `_check_availability` and `skip_if_missing` doesn't seem ideal, but also just `skip_if_missing` as a decorator is not ideal.

Reply:
You are both right, I did not follow the parameterization implication correctly.

I think we still want the decorator itself, for the `torch` use case in #1578 instead of the  `lyapunov` and `riccati` modules.

Reply:
This is now fixed: Tests are only skipped _after_ they fail with a `DependencyMissing`